### DISCUSSION TITLE
Simple sync follower question

### DISCUSSION THREAD

**Discussion by icewiz12** [NightscoutFoundation/xDrip#2304] [Q&A]
Hi all

The concept behind a sync follower seems simple so I thought I’d set up a factory reset Onn Android 9 tablet as a follower to my Samsung S10 xDrip+ for my nightstand. I have a Bluejay GTS as the collector and the S10 receiving data from that.  I want the Onn to follow the S10. 

I followed the excellent and dead simple steps on multiple tutorial websites (all the same) and I’m getting no readings on the Onn tablet even 12 hours later. I’ve done the Sync | Master, QR Code, Auto Confg | Received Parameters steps about 10 times.  

Any suggestions? Thx!

**Comment by Navid200** [comment]
You have readings in BlueJay?
And, you have readings in S10?

**Reply by Navid200** [reply]
If yes, can you post a screenshot of the xDrip sync settings page on the tablet after blacking out the sync key?

**Reply by icewiz12** [reply]
Yes, been using xDrip for many years and the GTS since it came out. All working seamlessly.  It's a simple mistake I'm making, but I can't find it.

**Reply by Navid200** [reply]
Does the sync key match the same on the master S10?
Is the last setting on the page disabled on both master and follower?

**Reply by Navid200** [reply]
https://navid200.github.io/xDrip/docs/xDrip-Sync-not-working.html

**Reply by icewiz12** [reply]
> If yes, can you post a screenshot of the xDrip sync settings page on the tablet after blacking out the sync key?

![Screen](https://user-images.githubusercontent.com/70183763/188153504-3cc16dc8-011b-478f-8594-64a024cfc186.jpg)

**Reply by icewiz12** [reply]
> Does the sync key match the same on the master S10? Is the last setting on the page disabled on both master and follower?

Yes, sync keys match.
The last setting is disabled.

**Reply by icewiz12** [reply]
Here's the S10 (minus the key stuff which is enabled and the correct key)

I've tried enabling the other functions and disabling them ... right now the rest are disabled.

![s10](https://user-images.githubusercontent.com/70183763/188153990-ccac9d73-bd92-4127-9f65-90899ff3de96.jpg)

**Reply by Navid200** [reply]
Does the tablet have Google Play Services on it?

**Reply by icewiz12** [reply]
> Does the tablet have Google Play Services on it?

Yes, first thing I checked. Says "installed".

Is there a way to turn error logging on for sync?  I'm not seeing anything in the log stream either positive or negative.

**Reply by Navid200** [reply]
What do you see on the system status page header "Followers" on your s10?

**Reply by Navid200** [reply]
The keys are case sensitive.  Are you sure the keys match?

Do both master and follower have access to the internet?

**Reply by icewiz12** [reply]
> What do you see on the system status page header "Followers" on your s10?

S10: XDrip+ Sync Group | We are the Master
Onn: xDrip+ Sync Group | Service State started 15:12, Slow wakeup 2 min | Wake Up Errors 65 | Slowest Wake up 4 min | Next Wakeup  4 min

**Reply by icewiz12** [reply]
> The keys are case sensitive. Are you sure the keys match?

Yes, both are upper case HEX and it was autoconfigured. 
> 
> Do both master and follower have access to the internet?

Yes. I have internet searched on both while this was occuring

**Reply by jonprogers** [reply]
> > What do you see on the system status page header "Followers" on your s10?
> 
> S10: XDrip+ Sync Group | We are the Master Onn: xDrip+ Sync Group | Service State started 15:12, Slow wakeup 2 min | Wake Up Errors 65 | Slowest Wake up 4 min | Next Wakeup 4 min

I use the follower functionality and I have extra information on the status screens of both the master and follower

Master:
Service state: We are the master
Network traffic: 2 mins ago
Follower: -follower name-
3 mins ago 100%

Follower:
Service state: started 08:57
Glucose data: 4 mins ago
Next wake up 4 mins
Network traffic 4 mins ago
Master: -master name-
4 mins ago 87%

It looks as though they don't know who they should be talking to. Mine are both signed into the same google account, is that the same for you? No clue if that is important!!

Good luck

**Reply by jamorham** [reply]
@icewiz12 can you try uninstalling xDrip from the follower and re-installing the latest nightly there and then try setup again. I assume you are using the mechanism like shown in my youtube video and scanning the QR code on the follower etc? Make sure your tablet has all updates and that google play store is working on it (as this would indicate google play services are likely properly installed)

**Reply by Navid200** [reply]
https://youtu.be/LcgjfbYcWkE

**Reply by icewiz12** [reply]
> @icewiz12 can you try uninstalling xDrip from the follower and re-installing the latest nightly there and then try setup again. I assume you are using the mechanism like shown in my youtube video and scanning the QR code on the follower etc? Make sure your tablet has all updates and that google play store is working on it (as this would indicate google play services are likely properly installed)

Please see proposed solution below. I tried many other variations, but once I got desert mode to work I knew it was possible. Then it was just a matter of unraveling as much as I could and seeing where it broke.  

Thanks as always!

**Comment by icewiz12** [comment]
Solution NOT Found:

UPDATE:  It is only working as Desert Mode even though the Master is not set for that.  It must be using my intranet.  It is still not working in cell or on broader internet or without Desert Mode set on the follower.  But it works well enough for me.


> I think you mean its not working? It feels like you've got data restrictions on one end or the other for google play services. That's really the only thing I can think of as this feature is normally simple to setup and reliable. Could either of your phones have something like that going on?

**Reply by jamorham** [reply]
Please check that following still works when not on the same wifi - that desert mode is not still active.

**Reply by icewiz12** [reply]
They are on different WiFi networks (one is a subnet of the other) and it appears to have worked while I was out on cell.  

Master has not been in Desert Mode since yesterday.

I just turned off Desert Mode on Follower and the updates stopped.  30 min later I turned Desert Mode back on Follower and it immediately (5 sec) updated.   It has continued to update.

The Follower was connected to a WiFi network that is a subnet of the primary network (different access points) the whole time.

**Reply by icewiz12** [reply]
Update:  It did NOT sync while I was out on cell.  It picked up again when I connected on wifi So seems to be working.

**Reply by jamorham** [reply]
I think you mean its not working? It feels like you've got data restrictions on one end or the other for google play services. That's really the only thing I can think of as this feature is normally simple to setup and reliable. Could either of your phones have something like that going on?

**Reply by icewiz12** [reply]
> I think you mean its not working? It feels like you've got data restrictions on one end or the other for google play services. That's really the only thing I can think of as this feature is normally simple to setup and reliable. Could either of your phones have something like that going on?

Sorry, meant it is working for how I need it to work. When I’m home it’s working and that’s all I need. 

It’s a factory reset tablet and a stock S10 that is refurbed so also “new”. The fact that my wifi is subdivided into subnets is a bit strange but all of the other Internet functions work including security cameras.  

I agree it should be dead simple. However without going thru desert mode and then leaving it on for the follower, it does not. I will test again today if it works on cell.

**Reply by Navid200** [reply]
We tell people how they can set up xDrip sync and when they do, it works even when they are not home and are on data rather than WiFi.  
There are children who go to school and while they are on the school bus, they have no access to WiFi.  But, as long as they are in a cell network, they have data and their parents can still see their glucose.

It's great that you are happy with your solution.
But, it will be wonderful if we can figure out what is causing the limitation.
So, please let us know if you find out.

**Reply by icewiz12** [reply]
UPDATE:  It is only working as Desert Mode even though the Master is not set for that.  It must be using my intranet.  It is still not working in cell or on broader internet or without Desert Mode set on the follower.  But it works well enough for me.

**Reply by Navid200** [reply]
Do you have a third Android device you can use for testing?
If yes, please set it up as a follower and see if it works.  What I mean by work is being able to follow with Desert Sync disabled.

**Reply by icewiz12** [reply]
> Do you have a third Android device you can use for testing? If yes, please set it up as a follower and see if it works. What I mean by work is being able to follow with Desert Sync disabled.

That's the FIRST thing I would have tried, but alas.  The rest of my household is IOS.  Even considered buying an extra tablet, but can't afford it right now, and wouldn't have a use for it separately.

If one of the devs want to try it, I can send a key and then change it.

**Reply by Navid200** [reply]
I can try to follow if you give me the key.
You can use Gitter and contact me at @Navid200.  After I respond, you can post a screenshot of the page containing the key.
And of course after the test, you can regenerate a new key.

**Reply by Navid200** [reply]
I was able to follow you using the key you provided.  
So, your master is working fine it seems.

**Comment by icewiz12** [comment]
Any issues with Android 9 stock?
It's as vanilla an install as I can have. No anti-virus, nothing.  The Onn tablet is pretty simple, and we know Desert Mode works.  
Very strange.
Are there any logs anywhere?
Did you simply put in the key as a follower? Or did you do something else?
    On Sunday, September 4, 2022 at 05:14:11 PM MDT, Navid ***@***.***> wrote:  
 
 


I was able to follow you using the key you provided.
So, your master is working fine it seems.

**Reply by Navid200** [reply]
I set the hardware data source to "xDrip+ Sync Follower".
Then, I went to the sync settings page and set the key.
That's it.

Is the tablet's time correct?  There is an Android setting that forces it to automatically sync with internet time.  Have you enabled that?

**Reply by icewiz12** [reply]
> I set the hardware data source to "xDrip+ Sync Follower". Then, I went to the sync settings page and set the key. That's it.
> 
> Is the tablet's time correct? There is an Android setting that forces it to automatically sync with internet time. Have you enabled that?

Time is set up properly. Internet time, etc.

I did your steps.  I installed xDrip+ from latest build on main page, I became xDrip+ Sync Follower, and I entered the key manually and triple checked it.  Nothing else.

Went to the System Status | Followers page and it says Service State: Stopped 17:35 

Went back to Hardware Source and hit xDrip Sync Follower and it started the service (per follower page).  So it appears I have to do that in reverse order with a new install.

No data immediately.  Waited for "next wakeup" on Follower page (started from 5).   It passed and no data was transferred.

So I agree it's something with the tablet (Android 9?) but I'm out of runway to figure out what.

Turned on Desert Mode and it immediately started working (without Hotspot ... I have them on the same WiFi AP now)

*shrug*

**Reply by icewiz12** [reply]
> I set the hardware data source to "xDrip+ Sync Follower". Then, I went to the sync settings page and set the key. That's it.
> 
> Is the tablet's time correct? There is an Android setting that forces it to automatically sync with internet time. Have you enabled that?

While I'm here, any way to make the font on the Home Screen bigger?  I think not (from looking at settings) but my use case is time and blood glucose reading, nothing else. Going to "always on" the screen overnight.  thx.

**Reply by Navid200** [reply]
You can change the colors:  https://navid200.github.io/xDrip/docs/Legend.html
But, not the font size.

**Comment by icewiz12** [comment]
Yes, first thing I checked. Says installed.
I don't see any messages on the S10 ... should there be debug messages or is there a way to turn them on?
    On Friday, September 2, 2022 at 12:29:57 PM MDT, Navid ***@***.***> wrote:  
 
 


Does the tablet have Google Play Services on it?

**Reply by Navid200** [reply]
This thread has become too old.
I have no idea which item you are talking about.  
Would you please do me two favors?

Would you please start a new thread?
Would you please not use email to post here?
